themes/default/content/blog/top-5-things-for-azure-devs-vm/index.md

```diff
@@ -0,0 +1,231 @@
+        subnet=network.SubnetArgs(id=net.subnets[0].id),
+        private_ip_allocation_method=network.IPAllocationMethod.DYNAMIC,
+        public_ip_address=network.PublicIPAddressArgs(id=public_ip.id),
+    )])
+
+```
+
+And here's our web server and index page using Python's simpeHTTPServer.
+
+```python
+init_script = """#!/bin/bash
+echo "Hello, World!" > index.html
+nohup python -m SimpleHTTPServer 80 &"""
+```
+
+Finally, here's how we create the virtual machine. We add the vm to the resource group and associate the `network_profile` to `network_iface` that we defined previously. We set the `vm_size`, set up the `os_profile` with the hostname and username and password. In the example, we create a `storage profile` and use Ubuntu for the image profile. The script ends by exporting the public IP address of the webserver.
+
+```python
+vm = compute.VirtualMachine(
+    "server-vm",
+    resource_group_name=resource_group.name,
+    network_profile=compute.NetworkProfileArgs(
+        network_interfaces=[
+            compute.NetworkInterfaceReferenceArgs(id=network_iface.id),
+        ],
+    ),
+    hardware_profile=compute.HardwareProfileArgs(
+        vm_size=compute.VirtualMachineSizeTypes.STANDARD_A0,
+    ),
+    os_profile=compute.OSProfileArgs(
+        computer_name="hostname",
+        admin_username=username,
+        admin_password=password,
+        custom_data=base64.b64encode(init_script.encode("ascii")).decode("ascii"),
+        linux_configuration=compute.LinuxConfigurationArgs(
+            disable_password_authentication=False,
+        ),
+    ),
+    storage_profile=compute.StorageProfileArgs(
+        os_disk=compute.OSDiskArgs(
+            create_option=compute.DiskCreateOptionTypes.FROM_IMAGE,
+            name="myosdisk1",
+        ),
+        image_reference=compute.ImageReferenceArgs(
+            publisher="canonical",
+            offer="UbuntuServer",
+            sku="16.04-LTS",
+            version="latest",
+        ),
+    ))
+
+combined_output = Output.all(vm.id, public_ip.name, resource_group.name)
+public_ip_addr = combined_output.apply(
+    lambda lst: network.get_public_ip_address(
+        public_ip_address_name=lst[1],
+        resource_group_name=lst[2]))
+export("public_ip", public_ip_addr.ip_address)
+```
+
+On the face of it, this seems more complex than using the Azure Portal. The critical thing to remember is that this process is reproducible, so if you require additional virtual machines, you can add a loop to create the necessary number of vms.
+
+But what about provisioning? Because this is code, you can write a [provisioning script](https://github.com/pulumi/examples/blob/master/aws-py-ec2-provisioners/provisioners.py) using Pulumi's [dynamic provider]({{< relref "/docs/tutorials/aws/aws-py-dynamicresource" >}}) that connects to the webserver, copies files, and executes commands. Here's a code snippet of a dynamic provider that copies files to your vm using scp.
```

Comment:
```suggestion
But what about provisioning? Because this is code, you can write a [provisioning script](https://github.com/pulumi/examples/blob/master/aws-py-ec2-provisioners/provisioners.py) using Pulumi's [dynamic provider]({{< relref "/docs/tutorials/aws/aws-py-dynamicresource" >}}) that connects to the webserver, copies files, and executes commands. Here's a code snippet of a dynamic provider that copies files to your VM using `scp`.
```